cmd/manager/main.go

```diff
@@ -257,5 +233,31 @@ func setFlagsFromEnv() {
 	if flags.identity.Name == "" {
 		flags.identity.Name = fmt.Sprintf("%s.%s", os.Getenv("POD_NAME"), os.Getenv("POD_NAMESPACE"))
 	}
+}
+
+const (
+	// ConfigMapKey is the key for mesh configuration data in the config map
+	ConfigMapKey = "mesh"
+)
+
+// fetch configuration from a config map
+func getMeshConfig(kube kubernetes.Interface) (*proxyconfig.ProxyMeshConfig, error) {
+	config, err := kube.CoreV1().ConfigMaps(flags.namespace).Get(flags.config, v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// values in the data are strings, while proto might use a different data type.
+	// therefore, we have to get a value by a key
+	yaml, exists := config.Data[ConfigMapKey]
+	if !exists {
+		return nil, fmt.Errorf("missing configuration map key %q", ConfigMapKey)
+	}
+
+	mesh := *envoy.DefaultMeshConfig
```

Comment:
Yes, this is important, otherwise, we'll be overriding default values. De-referencing to a stack variable does a  shallow copy of the pointer.